### PR TITLE
FEATURE: Add tag creation and tag editing to AI bot tools

### PR DESCRIPTION
**Pull request by Grubba27**
Follow-up to #42799 (stacked on it), doing for tags what that PR did for categories. The admin assistant's prompt says it helps with "tags", but its only tag write tool applies tags to a topic — it cannot create a tag ahead of time, rename one, or set a tag description.

- **`create_tag`** — creates a standalone tag with an optional description, normalizing the name through `DiscourseTagging.clean_tag` just like manual creation, and logs a `created_tag` staff action.
- **`edit_tag`** — renames an existing tag and/or changes its description (`can_edit_tag?`), logging the same `renamed_tag` staff action the tags controller writes.
- **`edit_tags` description fix** — the signature claimed "Adds tags to a topic. By default appends", but with `replace: true` it replaces the full tag set and removes omitted tags. The signature and `tags` parameter now say so; behavior is unchanged.

Both new tools require human approval and attribute the action to the approving moderator, matching the category tools. They sit inside the `tagging_enabled` gate in `all_available_tools`, so they disappear cleanly on sites without tagging.

### CODE REVIEW

**Comment by Grubba27** [comment]
Renamed the topic-tagging tool `edit_tags` → `edit_topic_tags` to remove the confusion with the new `edit_tag`, with a data migration renaming stored `EditTags` entries so existing agents keep the tool (same pattern as the `EditCategory` → `MoveTopic` migration in #42799).

**Review by ZogStriP — APPROVED** [review]

File: plugins/discourse-ai/lib/agents/tools/edit_topic_tags.rb [hunk 1]
```diff
@@ -3,11 +3,12 @@
 module DiscourseAi
   module Agents
     module Tools
-      class EditTags < Tool
+      class EditTopicTags < Tool
```
Comment (pmusaraj) [inline plugins/discourse-ai/lib/agents/tools/edit_topic_tags.rb]: I would recommend renaming this to `ChangeTopicTags`.

File: plugins/discourse-ai/lib/agents/tools/edit_tag.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          if !SiteSetting.tagging_enabled
+            return error_response(I18n.t("discourse_ai.ai_bot.edit_tag.errors.tagging_disabled"))
+          end
+
+          # Invisible tags report not-found so their names cannot be probed.
+          if tag.blank? || !guardian.can_see_tag?(tag)
+            return error_response(I18n.t("discourse_ai.ai_bot.edit_tag.errors.not_found"))
+          end
+
+          if parameters[:new_name].blank? && parameters[:description].blank?
```
Comment (pmusaraj) [inline plugins/discourse-ai/lib/agents/tools/edit_tag.rb]: We should make sure we allow clearing descriptions here. The tool treats an empty description as “not provided” here, it should allow setting it to empty, right?

File: plugins/discourse-ai/lib/agents/tools/edit_tag.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          fields
+        end
+
+        def perform_edit
+          if !guardian.can_edit_tag?(tag)
+            return error_response(I18n.t("discourse_ai.ai_bot.edit_tag.errors.not_allowed"))
+          end
+
+          previous_name = tag.name
+          tag.name = clean_new_name if parameters[:new_name].present?
```
Comment (pmusaraj) [inline plugins/discourse-ai/lib/agents/tools/edit_tag.rb:118]: We should validate `clean_new_name` here. Some names may result in clean values that are empty. For example a tag of "!!!" should not be allowed. We should add a test for this edge case.

**Review by pmusaraj — APPROVED** [review]
